### PR TITLE
Add doctests for HashMap's into_values and into_keys methods

### LINKED ISSUE #87591: Better doctests for HashMap's into_values and into_keys methods
The current doctests for a few methods don't really test anything when they could. The following changes would:

* improve the documentation for programmers by better showing what the results of the functions are
* bring the documentation in line with [the equivalent methods for `BTreeMap`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.into_keys), which do include tests like the following
* act as two free tests of those methods

---

[The current doctest for `HashMap`'s `into_keys` method](https://doc.rust-lang.org/std/collections/struct.HashMap.html#examples-31) can be improved to:

    use std::collections::HashMap;

    let mut map = HashMap::new();
    map.insert("a", 1);
    map.insert("b", 2);
    map.insert("c", 3);

    let mut vec: Vec<&str> = map.into_keys().collect();
    vec.sort_unstable();
    assert_eq!(vec, ["a", "b", "c"]);

---

[The current doctest for `HashMap`'s `into_values` method](https://doc.rust-lang.org/std/collections/struct.HashMap.html#examples-32) can be improved to:

    use std::collections::HashMap;

    let mut map = HashMap::new();
    map.insert("a", 1);
    map.insert("b", 2);
    map.insert("c", 3);

    let mut vec: Vec<i32> = map.into_values().collect();
    vec.sort_unstable();
    assert_eq!(vec, [1, 2, 3]);

---

I would submit a pull request, but my computer is broken. Could someone please make these changes?

### PR DESCRIPTION
**Pull request by ccqpein**
Fixes #87591

### CODE REVIEW

File: library/std/src/collections/hash/map.rs [hunk 1]
```diff
@@ -972,6 +972,8 @@ where
     /// map.insert("c", 3);
     ///
     /// let vec: Vec<&str> = map.into_keys().collect();
+    /// vec.sort_unstable();
```
Comment (lukaslueg) [inline library/std/src/collections/hash/map.rs:975]: A comment here to drive home the point that the `Vec` is in arbitrary order before the sort would be nice

**Comment by camelid** [comment]
Should this PR be marked as closing #87591?

File: library/std/src/collections/hash/map.rs (hunk 1 above)
Comment (ccqpein) [reply library/std/src/collections/hash/map.rs:975]: Done this in [commit](https://github.com/rust-lang/rust/pull/87598/commits/268a4f0bdbe25d8c47023c0604b0703095d65da9).

**Comment by ChaiTRex** [comment]
@ccqpein Can you please merge the pull request in your repository at https://github.com/ccqpein/rust/pull/1? It clarifies that the order of the elements produced by the iterators is arbitrary.

**Comment by ccqpein** [comment]
@ChaiTRex Thanks for your PR, have done!

**Comment by yaahc** [comment]
Looks great.

@bors r+

**Comment by JohnTitor** [comment]
We have the ["No-Merge Policy" in this repo](https://rustc-dev-guide.rust-lang.org/git.html#no-merge-policy), I think we should squash commits into one, could you?

@bors r-

**Comment by JohnCSimon** [comment]
Ping from triage: 
@ccqpein - can you please squash the commits and set the `S-waiting-on-review tag`? 
Thanks

**Comment by ccqpein** [comment]
@JohnCSimon I squash the commits but I am not sure I can add `labels` to PR.

**Comment by camelid** [comment]
You can do it like this: @rustbot label: -S-waiting-on-author +S-waiting-on-review

**Comment by ccqpein** [comment]
> You can do it like this: @rustbot label: -S-waiting-on-author +S-waiting-on-review

Thank you! Good to know, now I know how to this in my next PR to rustlang

**Comment by JohnTitor** [comment]
Thanks!
@bors r=yaahc

**Pull request merged**